reascripts/common/libs/PathUtil.lua

```diff
@@ -0,0 +1,80 @@
+--[[
+
+  PathUtil.lua - Path utilities for ReaScript
+
+]]--
+
+PathUtil = {}
+
+-- Apply an extension to a file path, if it doesn't already have one.
+-- Empty or nil path returns the empty string.
+PathUtil.apply_extension = function(filepath, extension)
+  if not filepath or #filepath < 1 then
+    return ""
+  end
+
+  local root, _, ext = filepath:match("([^%.]*)(%.?([^\\/%.]*))")
+
+  if not root or #root < 1 then
+    return ""
+  end
+
+  if #ext > 0 then
+    return filepath
+  end
+
+  return filepath .. '.' .. extension
+end
+
+-- Returns the given path if a full path, otherwise returns given path
+-- relative to the REAPER project resource directory.
+PathUtil.get_real_path = function(path_arg)
+  if PathUtil._is_full_path(path_arg) then
+    return path_arg
+  end
+
+  local project_path = reaper.GetProjectPath()
+  return project_path .. PathUtil._path_separator() .. path_arg
+end
+
+-- Returns the command to reveal the given path in the OS file explorer.
+-- Works for Windows and macOS. To quote a wise philosopher, "Linux is a mess."
+PathUtil.get_reveal_command = function(path_arg)
+  if PathUtil._is_windows() then
+    return 'explorer /select,"' .. path_arg .. '"'
+  elseif PathUtil._is_mac() then
+    return 'open -R "' .. path_arg .. '"'
+  else
+    return '[ -x /usr/bin/xdg-open ] && xdg-open "' .. path_arg .. '"'
+  end
+end
```

Comment:
doing some testing and noticing that:
- need to supply a full path to `open`; maybe the same for `explorer`?
- feeling like "and"-gating for linux isn't going to work 🤔🤔🤔

Reply:
just ran a **successful** test where i wrote:
```
/bin/[ -x /usr/bin/open ] && /usr/bin/open -R "/Users/mike/Documents/reaper/reaspeech-example/Media/test-export.json"
```
...to a `TempFile` and then did a no-wait `ExecProcess` with the argument:
```
/bin/bash <tempfile>
```

🤔🤔🤔